src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,45 +1,111 @@
 import { useState } from 'react';
 import { TextField } from '../TextField';
+import { Movie } from '../../types/Movie';
 
-export const NewMovie = () => {
-  // Increase the count after successful form submission
-  // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
+  const [count, setCount] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
+  const [isValidImg, setIsValidImg] = useState(true);
+  const [isValidImdb, setIsValidImdb] = useState(true);
+
+  function handleTextField(setText: (value: string) => void) {
+    const handleText = (value: string) => {
+      setText(value);
+    };
+
+    return handleText;
+  }
+
+  let cantSubmit = true;
+
+  if (title && imgUrl && imdbUrl && imdbId && isValidImg && isValidImdb) {
+    cantSubmit = false;
+  }
+
+  const handleSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+
+    onAdd({
+      title,
+      description,
+      imgUrl,
+      imdbUrl,
+      imdbId,
+    });
+
+    setTitle('');
+    setDescription('');
+    setImgUrl('');
+    setImdbUrl('');
+    setImdbId('');
+    setIsValidImg(true);
+    setIsValidImdb(true);
+    setCount((countPrev) => countPrev + 1);
+  };
 
   return (
-    <form className="NewMovie" key={count}>
+    <form
+      className="NewMovie"
+      key={count}
+      onSubmit={handleSubmit}
+    >
       <h2 className="title">Add a movie</h2>
 
       <TextField
         name="title"
         label="Title"
-        value=""
-        onChange={() => {}}
+        value={title}
+        setIsValidImg={setIsValidImg}
+        setIsValidImdb={setIsValidImdb}
+        onChange={handleTextField(setTitle)}
```

Comment:
It directly invokes the handleTextField function when the component renders. The onChange event expects a function reference that will be called when the input's value changes. 
```
onChange={() => ....}

Reply:
Not fixed. this function shouldn't be invoked on every render. BTW You can simplify it and get rid of the handleTextField method. 
```suggestion
        onChange={setTitle}
```